docs/pages/extensibility/ui.md

```diff
@@ -9,10 +9,46 @@ list: include
 ---
 Meshery UI has a number of extension points that allow for users to customize their experience with third-party plugins.
 
-## Designing custom plugins for Meshery UI
+## Designing Custom Components for Meshery Extension Point
+
+The Meshery extension points are the way to extend meshery and derive the more custom use-cases out of it. We already have an extension point called MeshMap. 
+Meshery can provide the extension point in various ways by providing the feature of custom-components. One of the example where these custom components are used is RJSF forms in meshery-extension
+
+### Extensibilty: RJSF Custom Component
+RJSFWrapperComponent are the customizations done on RJSF forms overriding the default behaviour of meshery-ui rjsf forms.
+The [Rjsf forms are wrapped](https://github.com/meshery/meshery/blob/0bc68d1cd0ba80a565afa68bce80899c22db9a2e/ui/components/MesheryMeshInterface/PatternService/RJSF.js#L66) under these component to receive the custom-props from Meshery-extension.
+```jsx
+<RJSFWrapperComponent {...props}>
+      <RJSFForm
+        isLoading={isLoading}
+        schema={schema}
+        data={data}
+        onChange={(e) => {
+          setData(e.formData)
+        }}
+        jsonSchema={jsonSchema}
+      />
+ </RJSFWrapperComponent> 
+```
+These props are received in the RJSF forms like this: [RJSF Component](https://github.com/meshery/meshery/blob/0bc68d1cd0ba80a565afa68bce80899c22db9a2e/ui/components/MesheryMeshInterface/PatternService/RJSF.js#L91) 
+
+#### Passing new custom prop to forms:
+
+ 1. Pass the new prop from the Meshery Extension in the RJSF Wrapper component used like this:
+
+ ```js
+function RJSFWrapperComponent(props) {
+  // Clone the child to pass in additional props
+  const children = React.cloneElement(props.children, {
+    ...(props.children?.props || {}),
+    customComponent: YOUR_NEW_CUSTOM_COMPONENT_OR_PROP
+  });
+
+  return children
+} 
+ ```
+Extract the props in the [RJSFForm Component](https://github.com/meshery/meshery/blob/0bc68d1cd0ba80a565afa68bce80899c22db9a2e/ui/components/MesheryMeshInterface/PatternService/RJSF.js#L91)
 
-_documentation coming soon..._
-<!-- Description of the process for designing plugins -->
```

Comment:
We can stay with it. The more documentation for creation of custom extensible point is required, so that newer extension points can be added.

Reply:
Good point. I added in a note to this effect...